library/item/item_rendering/src/main/java/org/quiltmc/qsl/item/rendering/mixin/client/ItemRendererMixin.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.item.rendering.mixin.client;
+
+import org.jetbrains.annotations.Nullable;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+
+import net.minecraft.client.font.TextRenderer;
+import net.minecraft.client.render.item.ItemRenderer;
+import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.item.ItemStack;
+
+import org.quiltmc.qsl.item.rendering.api.client.QuiltItemRenderingExtensions;
+
+@Mixin(ItemRenderer.class)
+public abstract class ItemRendererMixin {
+	@Unique private final ThreadLocal<MatrixStack> quilt$matrices = ThreadLocal.withInitial(MatrixStack::new);
+
+	@Shadow public float zOffset;
+
+	/**
+	 * @author QuiltMC
+	 * @reason Completely refactoring overlay rendering logic to use {@link QuiltItemRenderingExtensions}
+	 */
+	@Overwrite
```

Comment:
This overwrite is going to cause a lot of incompatibilities with fabric mods - is there any other way we can do it?

Reply:
I made it an `@Overwrite` simply because I couldn't be bothered to redo my old many-parts mixin at the time. I'll try to split it up into `@Inject`s and `@Redirect`s.

Reply:
...also, that third one is 100% copied from my old PR! It has the same comments as well.